AstarNetwork/swanky-cli
# Feature/init convert

Main feature is introducing a `--convert` flag to  `init` command. (#111 )
Also handled are #113 and #83 .

Some changes to how `init` works are introduced:
- all "tasks" now get pushed into the task queue and only executed after all the options are confirmed
- config is also gathered in the global configBuilder, and written as the last step

To test:
- have an existing project cloned to your filesystem (for example dex, or nft from https://github.com/swanky-dapps)
- run `swanky init proj_name --convert path_to_existing_project`
- it will check for Cargo.toml with `workspace` `members` and `exclude` fields
- it assumes `members` are contracts, and `exclude` are extra crates
- you can also specify manually
- when done, you still need to manually adjust import paths in the contracts' Cargo.toml

Note: the converted project is not expected to work out-of-the-box when converted, but should require minimum adjustments.
